### ISSUE TITLE
Scoped packages and Windows path separator

### ISSUE THREAD

**Issue by leops** [martpie/next-transpile-modules#25]
When using this plugin with namespaced modules such as `@org/module` the `transpileModule` property need to be rewritten as `@org[\\\\/]module` in order to work consistently on all platforms since the path separator on Windows is `\` instead of `/`.

This solutions seems somewhat fragile to me as the fact that the module name gets internally used in a RegExp (this is the reason why all the `\` have to be escaped twice, since the string literal gets parsed once by the JS engine then a second time by the RegExp constructor) should be an implementation detail of the library.

Maybe the RegExp generation code could automatically replace all the occurrences of `/` in the input with the `[\\\\/]` character set (or a properly escaped `path.sep`) ?

**Comment by martpie** [comment]
Are you saying you have to write that in `next.config.js`:

```
module.exports = withTm({
  transpileModules: [@org[\\\\/]module]
});
```
?

in order to transpile a scoped package on Windows?

**Comment by martpie** [comment]
Those regexp make things really complicated. I think we can switch to something more dev friendly, like functions. I will have to check Webpack's dock

**Comment by TxHawks** [comment]
I haven't done much on windows in a long while, but to the best of my knowledge it supports POSIX path separators since XP or Vista. If I'm correct, maybe there is not need to handle this at all.

[This stack overflow answer](https://stackoverflow.com/questions/8881806/file-separator-for-windows-and-unix) seem to support this, but since I'm not 100% on this, it would need to be checked.

If you want `next-transpile-modules` to handle this automatically, the `transpiledModules` array can be passed to something like this:

```js
function getCorrectModulePaths(modules:string[]):string[] {
  return modules.map(module => {
    const hasPosixPathSeperators = module.contains('/');
    const hasWindowsPathSeperators = module.contains('\\');
    if (!(hasPosixPathSeperators || hasWindowsPathSeperators)) {
      return module;
    }
    return path.join(...module
      .split(hasWindowsPathSeperators ? '\\' : '/')
      
  });
}
```

In the meanwhile, you can do:
```js
module.exports = withTm({
  transpileModules: [path.join('@org', 'module')]
});
```

Which will alway return the correct path separator. Not great, but better than `[\\\\/]`

**Comment by martpie** [comment]
Switching to globs is also a solution. I am currently investigating the different options

**Comment by martpie** [comment]
This looks promising but I cannot yet manage to make it work with the `@org/pack` path http://www.globtester.com/#p=eJzLy09Jjc%2FNTynNSS3WV9QozkgsSk2pSczLL8lILdKFcDX1tbT0tQBZ0A8E&r=eJyVzE0OQDAQhuGriD1zgWrcREY7UZTKzHB%2BYil%2BYvOt3uczvt8yF1GkyidUF3I7J0%2FNlPwaSYAJnYKSaDmIgaO25oPUiTtY0I3P7CIkIJMHWdu%2F5CV%2FwTgnDcTF7cm5OzqKZH8%3D&

**Comment by leops** [comment]
@TxHawks To be clear, the problem is not only to have "platform-insensitive" separators but also to escape them correctly for the internal RegExp created by this library (for instance on Windows `new RegExp(path.join('@types', 'node'))` will not match `@types\node` but `@types<line feed>ode` since the `\n` sequence gets parsed as a pseudo-class by the RegExp constructor)

**Comment by TxHawks** [comment]
Oh.... now I get it. Thanks.

How about windows understanding POSIX paths?

**Comment by leops** [comment]
Windows seems to understand POSIX paths without any kind of problem, at least I don't remember being particularly careful to path separators in my build scripts. But Webpack seems to do the matching of the RegExp specified in `config.module.rules.include` on the canonical path of the file being tested and not the POSIX one, so the file is a miss and doesn't get transpiled.

**Comment by martpie** [comment]
I will try to reproduce the bug at home on my windows laptop. I am not sure if Webpack's watch options compare the Regex to a POSIX path or a Win32 path.

**Comment by martpie** [comment]
@leops I cannot manage to reproduce the error, scoped packages are correctly transpiled on Win32. Can you provide a small repro and tell me your OS, your Next.js and your Node version?

**Comment by martpie** [comment]
Fixed in `1.3.0` (Next 7), and `2.2.0` (Next 8)
